eclair-core/src/main/scala/fr/acinq/eclair/wire/Onion.scala

```diff
@@ -228,13 +230,15 @@ object Onion {
     val expiry: CltvExpiry
     val paymentSecret: Option[ByteVector32]
     val totalAmount: MilliSatoshi
+    val keySend: Option[KeySend]
```

Comment:
I would directly put the preimage here to remove one unnecessary level of nesting:

```suggestion
    val paymentPreimage: Option[ByteVector32]
```

And in the `FinalTlvPayload` you would do:

```
override val paymentPreimage = records.get[KeySend].map(_.paymentPreimage) 
```

Reply:
Done